common/src/main/java/me/shedaniel/architectury/event/events/InteractionEvent.java

```diff
@@ -34,46 +34,132 @@
+         *
+         * @param player The player left clicking the block.
+         * @param hand The hand that is used.
+         * @param pos The position of the block in the level. Use {@link Player#getCommandSenderWorld()} to get the level.
+         * @param face The face of the block clicked.
+         * @return The event is canceled if anything else than {@link InteractionResult#PASS} is returned.
+         */
         InteractionResult click(Player player, InteractionHand hand, BlockPos pos, Direction face);
     }
     
     interface RightClickItem {
+        /**
+         * Called whenever a player is right clicking a block while holding an item.
+         * Equal to the forge {@code PlayerInteractEvent.RightClickItem} event and fabric's {@code UseItemCallback}.
+         *
+         * @param player The player right clicking the block.
+         * @param hand The hand that is used. You can get the {@link ItemStack} with {@link Player#getItemInHand(InteractionHand)}
+         * @return Whenever the return is not {@link InteractionResult#PASS}, the result value is used and the event is canceled.
+         */
         InteractionResultHolder<ItemStack> click(Player player, InteractionHand hand);
     }
     
     interface ClientRightClickAir {
+        /**
+         * Called when a player is right clicking the air.
+         * This is only fired on the client.
+         * Equal to the forge {@code PlayerInteractEvent.RightClickEmpty} event.
+         * 
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand The hand used.
+         */
         void click(Player player, InteractionHand hand);
     }
     
     interface ClientLeftClickAir {
+        /**
+         * Called when a player is left clicking the air.
+         * This is only fired on the client.
+         * Equal to the forge {@code PlayerInteractEvent.LeftClickEmpty} event.
+         *
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand The hand used.
+         */
         void click(Player player, InteractionHand hand);
     }
     
     interface InteractEntity {
+        /**
+         * Called when a player is right clicking on an entity.
+         * Equal to the forge {@code PlayerInteractEvent.EntityInteract} event.
+         * 
+         * @param player The player clicking the entity.
+         * @param entity Then entity the player clicks.
+         * @param hand The used hand.
+         * @return If the return value is not {@link InteractionResult#PASS}, event is cancelled and the used result is passed as return value.
+         */
         InteractionResult interact(Player player, Entity entity, InteractionHand hand);
     }
     
+    /**
+     * @deprecated use {@link BlockEvent#BREAK}
+     */
+    @Deprecated
     interface BlockBreak {
```

Comment:
Hmm, if this was replaced already by another event, why wasn't there a deprecation note before?